### DISCUSSION TITLE
Row grouping using Server side = true with Ajax Post method not working

### DISCUSSION THREAD

**Discussion by Ankitapatelit** [l-lin/angular-datatables#1699] [Q&A]
I am using below data table option in Angular application. I want to achieve Row grouping like this example using below code. https://datatables.net/examples/advanced_init/row_grouping.html . But it doesn't do row group. Any help is appreciated. Thank you. 

 **Type Script:**
  var groupColumn = 1;

  this.dtOptions = {             
      serverSide: true,
      processing: true,
      dom: 'QBfrtip',
      drawCallback: function (settings: any) {
        var api = this.api();
        var rows = api.rows({ page: 'current' }).nodes();
        var last: any = null;
        api.column(groupColumn, { page: 'current' })
          .data()
          .each(function (group: string, i: number) {
            if (last !== group) {
              $(rows).eq(i).before('<tr class="group"><td colspan="4">' + group + '</td></tr>');
              last = group;
            }
          });
      },
      ajax: (dataTablesParameters: any, callback: any) => {
        that.http
          .post<DataTablesResponse>(
            'https://xtlncifojk.eu07.qoddiapp.com/',
            dataTablesParameters, {}
          ).subscribe(resp => {
            that.persons = resp.data;
            callback({
              recordsTotal: resp.recordsTotal,
              recordsFiltered: resp.recordsFiltered,
              data: []
            });
          });
      },
      columns: [        { data: 'id' },        { data: 'firstName' },        { data: 'lastName' }]      
    };

**HTML:** 
<table datatable [dtOptions]="dtOptions" class="row-border hover">
    <thead>
      <tr>
        <th class="select-checkbox"><input id="checkBox" type="checkbox"  ></th>
        <th>ID</th>
        <th>First name</th>
        <th>Last name</th>
      </tr>
    </thead>
    <tbody *ngIf="persons?.length != 0">
      <tr *ngFor="let person of persons" [class.selectedRow]="rowIsSelected(person.id)">
        <td><input type="checkbox" class="form-check-input"></td>
        <td>{{ person.id }}</td>
        <td>{{ person.firstName }}</td>
        <td>{{ person.lastName }}</td>
      </tr>
    </tbody>
    <tbody *ngIf="persons?.length == 0">
      <tr>
        <td colspan="4" class="no-data-available">No data!</td>
      </tr>
    </tbody>
  </table>

**Comment by shanmukhateja** [comment]
Hi,

This is because your ajax callback's `data` property is empty. This indicates to datatables library that there is no data to render.

Please look at Angular Pipes or TemplateRef example in the documentation for recommended way.
